extensions/amp-instagram/0.1/amp-instagram.js

```diff
@@ -85,8 +86,10 @@ class AmpInstagram extends AMP.BaseElement {
     const image = this.win.document.createElement('amp-img');
     // This will redirect to the image URL. By experimentation this is
     // always the same URL that is actually used inside of the embed.
-    image.setAttribute('src', 'https://www.instagram.com/p/' +
-        encodeURIComponent(this.shortcode_) + '/media/?size=l');
+    const imgSrc = resolveUrlAttr('amp-img', 'src',
```

Comment:
I could see this possible but:

1. Not sure additional latency justifies it.
2. Can we actually make requests to this API from AMP? Does it need some sort of a key?
3. Is this API itself cookie-less? We can, of course, send a uncredentialed CORS request to ensure that it is, but I'm not sure what requirements are there.